src/routes/(app)/NotificationBell.svelte

```diff
@@ -1,32 +1,112 @@
 <script lang="ts">
   import Notification from "$lib/components/Notification.svelte";
+  import type { Notification as NotificationType } from "@prisma/client";
+  import { flip } from "svelte/animate";
+  // Enhance is used for actions without loading a page, similar to ajax
+  import { invalidate } from "$app/navigation";
+  import type { NotificationSchema } from "$lib/zod/schemas";
+  import type { SuperValidated } from "sveltekit-superforms";
+  import { superForm } from "sveltekit-superforms/client";
+  import { page } from "$app/stores";
+
+  export let notifications: NotificationType[];
+  export let deleteForm: SuperValidated<NotificationSchema>;
+
+  // Get the number of unread notifications, which is then used to indicate the user
+  $: unreadCount = notifications.filter((data) => data.readAt == null).length;
```

Comment:
This was previously a for loop which mutated an array. This meant reactivity didn't work. In svelte, as opposed to say react, a components script code is only run once, not on every "re-render". Re-renders in svelte work by finding dependencies, for example:
- If you declare a variable with let, it will keep track everywhere that is user
- Variables declared as `$: varName = ...` are called derived and svelte will keep track of which variables they use, the derived property will be updated whenever one of its dependencies changes.